src/logic/map_objects/tribes/productionsite.cc

```diff
@@ -1097,10 +1094,10 @@ void ProductionSite::set_default_anim(std::string anim) {
 void ProductionSite::update_actual_statistics(uint32_t duration, const bool produced) {
 	static const uint32_t duration_cap = 180 * 1000;  // This is highest allowed program duration
 	// just for case something went very wrong...
-	static const uint32_t entire_duration = 10 * 60 * 1000;
 	if (duration > duration_cap) {
 		duration = duration_cap;
 	}
+	static const uint32_t entire_duration = 5 * 60 * 1000;
```

Comment:
use a `constexpr uint32_t kStatsEntireDuration = ...;` outside the function body

Reply:
ok but just for my information could you explain this to me? Where shall I incorporate this and why? my c++ knowledge is very limited.

Reply:
Just above the function header:
~~~~

constexpr uint32_t kStatsEntireDuration = ...;
constexpr uint32_t kStatsDurationCap = ...;

void ProductionSite::update_actual_statistics(uint32_t duration, const bool produced) {
	if (duration > kStatsDurationCap) {
		duration = kStatsDurationCap;
	}
	const uint32_t past_duration = kStatsEntireDuration - duration;
	...
~~~~